pkg/generate/app/cmd/newapp.go

```diff
@@ -506,16 +526,18 @@ func (c *AppConfig) buildPipelines(components app.ComponentReferences, environme
 
 				// Append any exposed ports from Dockerfile to input image
 				if ref.Input().Uses.IsDockerBuild() {
-					exposed, ok := ref.Input().Uses.Info().Dockerfile.GetDirective("EXPOSE")
-					if ok {
-						if input.Info == nil {
-							input.Info = &imageapi.DockerImage{
-								Config: &imageapi.DockerConfig{},
+					if df, err := ref.Input().Uses.Info().Dockerfile.Dockerfile(); err == nil {
+						exposed, ok := df.GetDirective("EXPOSE")
```

Comment:
There is no code change required for that though since Config is already
set.  I suppose we should unify these code paths, but I don't know that
it's a clear factoring yet.

On Thu, Sep 24, 2015 at 9:32 AM, Cesar Wong <notifications@github.com>
wrote:

> In pkg/generate/app/cmd/newapp.go
> <https://github.com/openshift/origin/pull/4717#discussion_r40318391>:
>
> > @@ -506,16 +526,18 @@ func (c *AppConfig) buildPipelines(components app.ComponentReferences, environme
> >
> >  				// Append any exposed ports from Dockerfile to input image
> >  				if ref.Input().Uses.IsDockerBuild() {
> > -					exposed, ok := ref.Input().Uses.Info().Dockerfile.GetDirective("EXPOSE")
> > -					if ok {
> > -						if input.Info == nil {
> > -							input.Info = &imageapi.DockerImage{
> > -								Config: &imageapi.DockerConfig{},
> > +					if df, err := ref.Input().Uses.Info().Dockerfile.Dockerfile(); err == nil {
> > +						exposed, ok := df.GetDirective("EXPOSE")
>
> sorry we do:
> https://github.com/openshift/origin/blob/1b9f255f61d66d97e644cd283612a816f46aa550/pkg/generate/app/app.go#L369
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4717/files#r40318391>.
>